### PR TITLE
🧪 Enable experiment to upgrade amp-fit-text 0.1 to 1.0

### PR DESCRIPTION
**Pull request by caroqliu**
Related to #32523 and #28281

Note:
- Importing `BaseElement` is preferred to `AmpFitText` because we do not need to assert the `"bento"` experiment is on (we turn this on manually). Open question: Could the `registerExtension` in `1.0` also potentially collide with the one in `0.1`?
- CSS is not copied from 1.0 to 0.1 as these are pre-upgrade styles that should not be needed with boilerplate on the page.

### CODE REVIEW

File: extensions/amp-fit-text/0.1/amp-fit-text.js [hunk 1]
```diff
@@ -54,6 +56,27 @@ class AmpFitText extends AMP.BaseElement {
...
+    return !!BENTO_AUTO_UPGRADE;
+  }
+
+  /** @override */
+  upgradeCallback() {
+    if (
+      this.isInBentoUpgradeExp() &&
+      typeof Element.prototype.attachShadow == 'function'
+    ) {
+      toggleExperiment(this.win, 'bento', true, false);
```
Comment (dvoytenko) [inline extensions/amp-fit-text/0.1/amp-fit-text.js:74]: Can we avoid this?

File: extensions/amp-fit-text/0.1/amp-fit-text.js [hunk 2]
```diff
@@ -54,6 +56,27 @@ class AmpFitText extends AMP.BaseElement {
     this.textContent_ = '';
   }
 
+  /**
+   * @visibleForTesting
+   * @return {boolean}
+   */
+  isInBentoUpgradeExp() {
```
Comment (dvoytenko) [inline extensions/amp-fit-text/0.1/amp-fit-text.js:63]: Not sure we need to have a test for it, tbh. IMHO it's fine to inline.

**Review by dvoytenko — CHANGES_REQUESTED** [review]

File: extensions/amp-fit-text/0.1/amp-fit-text.js (hunk 1 above)
Comment (caroqliu) [reply extensions/amp-fit-text/0.1/amp-fit-text.js:74]: Removed.

File: extensions/amp-fit-text/0.1/amp-fit-text.js (hunk 2 above)
Comment (caroqliu) [reply extensions/amp-fit-text/0.1/amp-fit-text.js:63]: Inlined

File: extensions/amp-fit-text/0.1/amp-fit-text.js [hunk 3]
```diff
@@ -54,6 +55,18 @@ class AmpFitText extends AMP.BaseElement {
     this.textContent_ = '';
   }
 
+  /** @override */
+  upgradeCallback() {
+    if (
+      // eslint-disable-next-line no-undef
```
Comment (dvoytenko) [inline extensions/amp-fit-text/0.1/amp-fit-text.js:61]: I think eslint not needed anymore here. See https://github.com/ampproject/amphtml/pull/32538/files

**Review by dvoytenko — APPROVED** [review]

File: extensions/amp-fit-text/0.1/amp-fit-text.js (hunk 3 above)
Comment (caroqliu) [reply extensions/amp-fit-text/0.1/amp-fit-text.js:61]: Running `gulp lint` yields:
```
extensions/amp-fit-text/0.1/amp-fit-text.js
  61:7  error  'BENTO_AUTO_UPGRADE' is not defined  no-undef
```

Comment (dvoytenko) [reply extensions/amp-fit-text/0.1/amp-fit-text.js:61]: /cc @rsimha

Comment (rsimha) [reply extensions/amp-fit-text/0.1/amp-fit-text.js:61]: This is a case where both of you are correct 😃 The `eslint-disable` directive won't be necessary if you add `BENTO_AUTO_UPGRADE` to this file:

https://github.com/ampproject/amphtml/blob/327d021b18ad902711370dae815087fc70a29889/build-system/global-configs/experiments-const.json#L1-L4

Comment (caroqliu) [reply extensions/amp-fit-text/0.1/amp-fit-text.js:61]: Got it, thanks!

**Comment by caroqliu** [comment]
Two additional notes: 
- `"WEB"` is not an exclusive format, so in order to not impact ads/email/stories documents which use `amp-fit-text`, I've set the environment to `"AMP"` and the upgrade condition to include `isAmphtml()`
- I'm conditionally skipping e2e tests for amp-fit-text:0.1 if the experiment is enabled. -- This is because when 0.1 auto upgrades to 1.0, the component is executed via shadow dom and the e2e tests for 0.1 which assert styles in the light dom are expected to fail. I'm not sure how to reproduce this locally on my machine after talking with @estherkim, so I'm prepared to follow up if this breaks the master e2e build in Travis.

**Pull request merged**